test/ng/compileSpec.js

```diff
@@ -7815,4 +7817,125 @@ describe('$compile', function() {
         expect(element.hasClass('fire')).toBe(true);
       }));
   });
+
+  describe('element replacement', function() {
+    it('should broadcast $destroy only on removed elements, not replaced', function() {
```

Comment:
Does anyone know if this is actually intentional? Why is this done? It seems like the replaced node was never `.remove()`ed so this was always the case. But when multi-element directives were added (e46100f7097d9a8f174bdb9e15d4c6098395c3f2) they used `.remove()` which does fire `$destroy`.

Should `$destroy` actually be fired on the replaced node as well?